### DISCUSSION TITLE
How to handle non-atomic migrate + code release?

### DISCUSSION THREAD

**Discussion by jyasskin** [prisma/prisma#17074] [Q&A]
TL;DR: How can Prisma help me ensure that a migration is safe to perform on a running system?

When upgrading a production system, we'll have v1 of the code written against v1 of the schema, trying to upgrade to v2 of the code written against v2 of the schema. While [some DBs make it possible to upgrade the schema in a transaction](https://github.com/prisma/prisma/issues/8080), the serving code almost always has a period when both versions of the code are running at the same time.

Does Prisma have any advice or safeguards to help us ensure that either v1 code can run against the v2 schema (in which case we'd run `prisma migrate deploy` before starting to roll out the new code) or that v2 code can run against the v1 schema (in which case we'd migrate after rolling out the new code)?

**Comment by nurul3101** [comment]
Hello @jyasskin 👋 

One idea I can think of is to use the [prisma migrate diff](https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-diff) command to preview the SQL queries that will be executed as part of the migration. This can help you understand the impact of the migration and identify any potential issues before applying it to your database.

**Reply by jyasskin** [reply]
It's straightforward to read the migrations directory to see what how the tables are going to change, but it's harder to check that all my app's uses of `prisma.*` are actually compatible with the new schema. Automating those checks is why Prisma's TypeScript support is so valuable.

**Comment by anitacung** [comment]
👋🏻 Hi @jyasskin,

I'm going through pending Github Discussions to check if you need any help. Have you had the chance to take a look at Nurul's recommendation? 

Let us know if this issue is still open!

**Reply by jyasskin** [reply]
I think it's still open. It's not super urgent—I can test the previous release's code against a staging version of the upgraded database if I have to—but it'd be nice to have a more systematic way to find or prevent issues.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Reply by jyasskin** [reply]
I'm still interested in a solution to this, but I can't reopen the discussion.
